packages/framework/presence/src/test/notificationsManager.spec.ts

```diff
@@ -3,16 +3,268 @@
+
+import { EventAndErrorTrackingLogger } from "@fluidframework/test-utils/internal";
+import { useFakeTimers, type SinonFakeTimers, spy } from "sinon";
+
+import { Notifications, type ISessionClient } from "../index.js";
+import type { IEphemeralRuntime } from "../internalTypes.js";
 import type { IPresence } from "../presence.js";
+import type { createPresenceManager } from "../presenceManager.js";
+
+import { MockEphemeralRuntime } from "./mockEphemeralRuntime.js";
+import { assertFinalExpectations, prepareConnectedPresence } from "./testUtils.js";
+
+/**
+ * A map of presence signals used for testing, indexed by a friendly string.
+ */
+const knownSignals = new Map<string, Parameters<IEphemeralRuntime["submitSignal"]>>([
+	[
+		"PresenceJoin",
+		[
+			"Pres:DatastoreUpdate",
+			{
+				"sendTimestamp": 1010,
+				"avgLatency": 10,
+				"isComplete": true,
+				"data": {
+					"system:presence": {
+						"clientToSessionId": {
+							"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+						},
+					},
+				},
+			},
+		],
+	],
+
+	[
+		"setNewId42",
+		[
+			"Pres:DatastoreUpdate",
+			{
+				"sendTimestamp": 1020,
+				"avgLatency": 10,
+				"data": {
+					"system:presence": {
+						"clientToSessionId": {
+							"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+						},
+					},
+					"n:name:testNotificationWorkspace": {
+						"testEvents": {
+							"sessionId-2": {
+								"rev": 0,
+								"timestamp": 0,
+								"value": { "name": "newId", "args": [42] },
+								"ignoreUnmonitored": true,
+							},
+						},
+					},
+				},
+			},
+		],
+	],
+]);
 
 describe("Presence", () => {
 	describe("NotificationsManager", () => {
-		/**
-		 * See {@link checkCompiles} below
-		 */
-		it("API use compiles", () => {});
+		// Note: this test setup mimics the setup in src/test/presenceManager.spec.ts
+		let runtime: MockEphemeralRuntime;
+		let logger: EventAndErrorTrackingLogger;
+		const initialTime = 1000;
+		let clock: SinonFakeTimers;
+		let presence: ReturnType<typeof createPresenceManager>;
+
+		beforeEach(() => {});
+
+		before(async () => {
+			clock = useFakeTimers();
+		});
+
+		beforeEach(() => {
+			logger = new EventAndErrorTrackingLogger();
+			runtime = new MockEphemeralRuntime(logger);
+
+			// TODO: runtime.connected has been hacked in past to lie about true connection.
+			// This will need to be updated to an alternate status provider.
+			// runtime.connected = true;
+
+			clock.setSystemTime(initialTime);
+
+			// Set up the presence connection, which will create an expected signal, so add that signal to the expected list
+			// as well
+			presence = prepareConnectedPresence(runtime, "sessionId-2", "client2", clock, logger);
+			runtime.signalsExpected.push(
+				// The first signal, the join signal, is from the prepareConnectedPresence call
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				knownSignals.get("PresenceJoin")!,
+			);
```

Comment:
This doesn't add up for me.
1) `prepareConnectedPresence` already takes care of pushing the signal expected from it (see https://github.com/tylerbutler/FluidFramework/blob/presence-notification-events/packages/framework/presence/src/test/testUtils.ts#L96)
2) signalsExpected should always be added before they might happen (the array is shortened as events happen)

I could see that `prepareConnectedPresence` doesn't submit signal here because `runtime.connected` was not set to `true`. Something though must convince PresenceManager that it is connected at some point for signals to get submitted. I don't see later code explicit changing that state.
Hmm...

Reply:
I debugged. What is happening is that there is a Join signal process just below. That Join requests that the local client ("client2") respond. The signal expected is the one sent to satisfy that request. It doesn't look like anything relies on there being another client present. So, could just remove the expected signal push and the processSignal below.

Reply:
Thanks - I cleaned things up quite a bit in the latest version. Working on the core code changes now.